docs/src/templates/pages/documentation/js/writing-tests.hbs

```diff
@@ -0,0 +1,304 @@
+- **DRY**: Don't Repeat Yourself, avoid code duplication
+- **Single responsibility**: each object/function must focus on a single task
+- Keep a **single level of abstraction** in the same component (for example, do not mix business logic with lower-level technical details in the same method)
+- **Minimize dependencies** between components: encapsulate, interchange less information between components
+- **Support configurability** rather than hard-coding, this prevents from having to replicate the exact same environment when testing (e.g.: markup)
+- Avoid global mutable state
+
+
+<a name="guidelines"></a><h2 class="sg-sectionHeading">Guidelines</h2>
+
+The goal of these guidelines is to make your tests:
+
+- **Readable**
+- **Maintainable**
+- **Trustworthy**
+
+These are the 3 pillars of good unit testing.
+
+
+<h3 class="sg-sectionHeading">Name your tests properly</h3>
+
+Tests names should be concise, explicit, descriptive and in correct English. Read the output of the spec runner and verify that it is understandable! Keep in mind that someone else will read it too. Tests can be the live documentation of the code.
+
+**Bad:**
+
+```javascript
+it('invalid selector works', () => {
+    // ..
+});
+```
+
+**Good:**
+
+```javascript
+it('showElements does not affect hiddenClass if invalid selector is passed in', () => {
+    // ..
+});
+```
+
+
+<h3 class="sg-sectionHeading">Don't comment tests</h3>
+
+Never. Ever. Tests have a reason to be or not.
+
+Don't comment them because they are too slow, too complex or produce false negatives. Instead, make them fast, simple and trustworthy. If not, remove them completely.
+
+
+<h3 class="sg-sectionHeading">Avoid logic in your tests</h3>
+
+Always use simple statements. Loops and conditionals must not be used. If they are, you add a possible entry point for bugs in the test itself:
+
+- Conditionals: you don't know which path the test will take
+- Loops: you could be sharing state between tests
+
+Write a test for each type of sanitization. It will give a nice output of all possible cases, improving readability and maintainability.
+
+```javascript
+const invalidModalTypes = [null, undefined];
+invalidModalTypes.forEach(modalType => {
+    it(`type option defaults to "full" when invalid value of "${modalType}" is set`, () => {
+        // Act
+        const options = { modalType };
+
+        // Act
+        const modal = new MegaModal(options);
+
+        // Assert
+        expect(modal.options.modalType).toBe('full');
+    });
+});
+```
+
+```javascript
+// Example of well separated tests
+
+it('default value is set when passing null', () => {
+    // Act
+    const modalType = null;
+    const options = { modalType };
+
+    // Act
+    const modal = new MegaModal(options);
+
+    // Assert
+    expect(modal.options.modalType).toBe('full');
+});
+
+it('default value is set when passing undefined', () => {
```

Comment:
Possibly – this is just an excerpt copied from another doc about JS testing.  

Not too concerned as the aim is to highlight the separation of the tests over describe grouping here.  Happy for someone to edit it if we think it could be represented better though.